File: Assets/JumpScareTrig.js
var hero : GameObject;
var scare : GameObject;
var played = false;
var trig = false;

function Start () {
	trig = false;
	scare.GetComponent.<Renderer>().enabled = false;
}

function OnTriggerEnter (other : Collider) {
	trig = true;
}

function Update () {
    if (trig == true) {
        scare.transform.position = new Vector3(hero.transform.position.x, hero.transform.position.y, hero.transform.position.z -10);
		scare.GetComponent.<Renderer>().enabled = true;
		removeovertime ();
		makehimscream ();
	}
}

function removeovertime () {
	yield WaitForSeconds (0.8);
	scare.GetComponent.<Renderer>().enabled = false;
	Destroy(this.gameObject);
	
}

function makehimscream () {
	if (!played) {
		played = true;
		GetComponent.<AudioSource>().Play();
	}
}

File: Assets/Standard Assets/JumpScareTrig.cs
using UnityEngine;
using System.Collections;
using UnityStandardAssets.Characters.ThirdPerson;

public class JumpScareTrig : MonoBehaviour
{
    public GameObject Player;
    public GameObject ScarePane;
    public GameObject AiController;
    public bool played = false;
    public bool trig = false;

    void Start()
    {
        trig = false;
        ScarePane.GetComponent<Renderer>().enabled = false;
    }

    void OnTriggerEnter(Collider other)
    {
        if(other.tag.Equals("Hero"))
        {
            if (!trig)
            {
                trig = true;
                ScarePane.transform.position = new Vector3(Player.transform.position.x, Player.transform.position.y, Player.transform.position.z - 10);
                ScarePane.GetComponent<Renderer>().enabled = true;
                GetComponent<AudioSource>().Play();
                var control = AiController.GetComponent<AICharacterControl>();
                control.characterState.Scare(control);
                Destroy(ScarePane, 1f);
            }
        }
    }
}